content/200-concepts/100-components/02-prisma-client/260-client-extensions/index.mdx

```diff
@@ -0,0 +1,170 @@
+
+Every extended client operates independently, in an isolated instance. It cannot conflict with other extended clients or the original client.
+
+You can also combine two or more extensions into one extension, if you want the functionality from both extensions to apply in the same client instance.
+
+### Examples of isolated client instances
+
+Example uses of isolated client instances include the following:
+
+- Implement row-level security (RLS), where each http request has its own client with its own RLS extension, customized with session data. This can keep each user entirely separate, each in a separate client.
+- Add a `user.current()` method for the `user` model to get the current logged-in user
+- Enable more verbose logging for requests if a debug cookie is set
+- Attach a unique request id to all logs so you can correlate them later
+- Remove a `delete` method from models unless the application calls the admin endpoint and the user has the necessary priviliges
+
+### Declare multiple extensions
+
+The following example declares two separate extensions and one combined extension:
+
+```ts
+// First of all, store your original prisma client in a variable as usual
+const prisma = new PrismaClient()
+
+// Declare extension A
+const extendsA = prisma.$extends(extensionA)
+
+// Declare extension B
+const extendsB = prisma.$extends(extensionB)
+
+// Declare an extension that is a combination of A and B
+const extendsAB = prisma.$extends(extensionA).$extends(extensionB)
+```
+
+When you combine two or more extensions, then the last extension declared takes precedence in any conflict. In the example above, suppose there is a method called `getName` in `extendsA` and a method called `getName` in `extendsB`. When you call `extendsAB.getName`, then client uses `getName` as defined in `extendsB`.
+
+### Extensions versus middlewares
+
+Extensions differ from [middlewares](/concepts/components/prisma-client/middleware#performance-and-appropriate-use-cases) in the following ways:
+
+- If you use two or more middlewares at the same time, then they all apply globally to the same client
+- Extensions are isolated: the functionality in each one is separate, unless you deliberately combine them
+- All extended clients share the same connection pool, but their functionality is separate
+- In the RLS example given above, with extensions each user can be in an entirely separate client. With middlewares, all users are active in the same client.
+
+## Share Prisma Client extensions
+
+You can share Prisma Client extensions with other users, either as packages or as code snippets. To create an extension that is more likely to work for other users, use the `$all` prefix, for example [`$allModels`](/concepts/components/prisma-client/client-extensions/client-extensions-model#add-a-custom-method-to-all-models-in-your-schema). This extension applies generally, not just to your specific schema.
```

Comment:
This is weird - as we do not know anything about this on this page (yet). (First time we even see the level below `model:` is this example)

Reply:
Any example we give for sharing extensions will be more advanced than what we've covered on this page so far. I think the solution for this is to split out the sharing section to another page. Then we don't get narrative dissonance.